build_tools/cmake/iree_e2e_generated_runner_test.cmake

```diff
@@ -251,9 +243,13 @@ function(iree_single_backend_e2e_runner_test)
 
   list(APPEND _GENERATOR_STANDARD_FLAGS "--output_${_RULE_TEST_TYPE}_mlir=${_TESTS_SRC}")
   list(APPEND _GENERATOR_STANDARD_FLAGS "--output_calls_mlir=${_CALLS_SRC}")
-  if(_RULE_TARGET_CPU_FEATURES)
-    list(APPEND _GENERATOR_STANDARD_FLAGS "--requirements=${_RULE_TARGET_CPU_FEATURES}")
-  endif()
+  foreach(_COMPILER_FLAG IN LISTS _RULE_COMPILER_FLAGS)
+    set(_CPU_FEATURES_REGEX "^--iree-llvmcpu-target-cpu-features=")
+    if (_COMPILER_FLAG MATCHES "${_CPU_FEATURES_REGEX}")
+      string(REGEX REPLACE "${_CPU_FEATURES_REGEX}" "" _CPU_FEATURES "${_COMPILER_FLAG}")
+      list(APPEND _GENERATOR_STANDARD_FLAGS "--requirements=${_CPU_FEATURES}")
+    endif()
+  endforeach()
```

Comment:
This is fine for now / in this repo.

> > > I have followed your suggestion to drop `TARGET_CPU` and `TARGET_CPU_FEATURES`. While doing do, I found the place which had originally motivated them: this is where we pass a `--requirements` flag to the test runner so that it can check at runtime that the host CPU supports the expected features. Now that there isn't a separate `TARGET_CPU_FEATURES` field anymore, we resort to parsing the `COMPILER_FLAGS`, which is not too bad and is a tiny detail, so it is the right call to deal with it rather than have those `TARGET_CPU` and `TARGET_CPU_FEATURES` flags everywhere.
> 
> Ah. That's on my list to remove too: https://github.com/iree-org/iree-test-suites/blob/3a0ea13cbdc954365d653a48cc99cc63a9ff09b3/linalg_ops/matmul/generate_e2e_matmul_tests.py#L529-L531
> 
> ```
>     # TODO(scotttodd): drop this and whatever logic in the test tool used it
>     #     multiple backends should be able to use the same input IR, so the
>     #     input IR shouldn't need things like CPU features in it
> ```

(Over in [iree-test-suites](https://github.com/iree-org/iree-test-suites)) I think I also want to push this logic to the leaves, similar to how we have `if(IREE_HIP_TEST_TARGET_CHIP)` controlling defining and running tests for the ROCm backend + HIP driver, we can check explicit options like `IREE_TEST_SUITES_INCLUDE_ARM_SME_TESTS` or check automatically/programmatically for support.

Basically, each set of tests should make it obvious how the tests are enabled. The existing logic, which you are updating here, hides some of that decision making in helper functions. For remote targets like Android devices and local targets like GPUs, we can't as easily enumerate devices and supported features at configure time, and I don't want to special case any backend in the helper functions.